src/basic/env-file.c

```diff
@@ -494,6 +495,25 @@ int load_env_file_pairs(FILE *f, const char *fname, char ***ret) {
         return 0;
 }
 
+int load_env_file_pairs_fd(int fd, const char *fname, char ***ret) {
+        _cleanup_close_ int fd_ro = -EBADF;
+        _cleanup_fclose_ FILE *f = NULL;
+
+        assert(fd >= 0);
+
+        fd_ro = fd_reopen(fd, O_CLOEXEC | O_RDONLY);
+        if (fd_ro < 0)
+                return fd_ro;
+
+        f = fdopen(fd_ro, "re");
+        if (!f)
+                return -errno;
+
+        TAKE_FD(fd_ro);
```

Comment:
please rebase on #26952 and use the new helper here.

also, one existing use of load_env_file_pairs() should be ported over to the new load_env_file_pairs_fd(), it's trivial. Please do so right-away when adding the helper.